packages/rs-sdk/src/platform/transition/put_contract.rs

```diff
@@ -10,14 +10,14 @@ use dpp::identity::signer::Signer;
 use dpp::identity::{IdentityPublicKey, PartialIdentity};
 use dpp::state_transition::data_contract_create_transition::methods::DataContractCreateTransitionMethodsV0;
 use dpp::state_transition::data_contract_create_transition::DataContractCreateTransition;
-use dpp::state_transition::proof_result::StateTransitionProofResult;
 use dpp::state_transition::StateTransition;
 
 use super::broadcast::BroadcastStateTransition;
+use super::waitable::Waitable;
 
 #[async_trait::async_trait]
 /// A trait for putting a contract to platform
-pub trait PutContract<S: Signer> {
+pub trait PutContract<S: Signer>: Waitable {
```

Comment:
In other cases you don't add super trait so you need to use `<Self as Waitable>::`. I think this approach is cleaner